### PR TITLE
Aggregate fee tests

### LINKED ISSUE #823: Adjust `VaultLiquidityFees` test
It's still using protocol / creator fees, which doesn't apply anymore.
Use single aggregate fee percentage.

### PR DESCRIPTION
**Pull request by EndymionJkb**
# Description

We had a legacy helper in `BaseVaultTest` (and a duplicate in one of the test files) that computed aggregate percentages from their components. It had (documented) precision issues, which were resolved in the latest `ProtocolFeeController` code.

This makes minimal changes to remove these helpers and use the `ProtocolFeeController` itself for aggregate computation, so that it's consistent with what the system does. For the fuzz tests, there were still precision issues trying to set each to its full range, so in that case I had to fuzz the aggregate fee directly (bounding it, then manually adjusting for precision).

A more extensive change would be to always use the aggregate percentage natively, and not try to compute it.

## Type of change

- [ ] Bug fix 
- [ ] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [X] Code refactor / cleanup
- [ ] Optimization: [ ] gas / [ ] bytecode
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [X] The diff is legible and has no extraneous changes
- [X] Complex code has been commented, including external interfaces
- [ ] Tests have 100% code coverage
- [X] The base branch is either `main`, or there's a description of how to merge

## Issue Resolution

Resolves #823 .

### CODE REVIEW

File: pkg/vault/test/foundry/unit/VaultUnitSwap.t.sol [hunk 1]
```diff
@@ -43,9 +44,12 @@ contract VaultUnitSwapTest is BaseTest {
     uint256[] decimalScalingFactors = [uint256(1e18), 1e18];
     uint256[] tokenRates = [uint256(1e18), 2e18];
 
+    IProtocolFeeController feeController;
+
     function setUp() public virtual override {
         BaseTest.setUp();
         vault = IVaultMock(address(VaultMockDeployer.deploy()));
+        feeController = vault.getProtocolFeeController();
```
Comment (jubeira) [inline pkg/vault/test/foundry/unit/VaultUnitSwap.t.sol:52]: Maybe we can put this in the base layer, and make `_getAggregateFeePercentage` use it?

Comment (jubeira) [reply pkg/vault/test/foundry/unit/VaultUnitSwap.t.sol:52]: (or just keep the code as is now, but store `feeController` reference in the base test to reduce setup in every test)

**Review by jubeira — APPROVED** [review]
LGTM, thanks @EndymionJkb !

How about storing the `feeController` in the base test? That would reduce a few setup lines here and there; it could be useful to have.

File: pkg/vault/test/foundry/unit/VaultUnitSwap.t.sol (hunk 1 above)
Comment (EndymionJkb) [reply pkg/vault/test/foundry/unit/VaultUnitSwap.t.sol:52]: Moved it up to the BaseVaultTest

**Pull request merged**